src/messages/mod.rs

```diff
@@ -625,6 +625,8 @@ pub enum MessageContent {
     UserMessagePart {
         /// The hash of this user message.
         hash: Digest256,
+        /// The unique message ID of this user message.
+        msg_id: MessageId,
```

Comment:
as the msg_id has been moved into the UserMessagePart,
can the msg_id in Response/Request be removed?
or this will be a too big change and shall leave to be done later on?